packages/deployments/contracts/contracts/messaging/RootManager.sol

```diff
@@ -141,6 +143,20 @@ contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainInde
     emit ConnectorRemoved(_domain, _connector, domains, connectors, msg.sender);
   }
 
+  /**
+   * @notice Removes (effectively blacklists) a given (fraudulent) root from the queue of pending
+   * inbound roots.
+   * @dev The given root does NOT have to currently be in the queue. It isn't removed from the queue
+   * directly, but instead is filtered out when dequeuing is done for the sake of aggregation.
+   * @dev Can only be called by the owner when the protocol is paused.
+   *
+   * @param _root The root to be discarded.
+   */
+  function discardRoot(bytes32 _root) public onlyOwner whenPaused {
```

Comment:
@csanuragjain See method here added as the interface to discard (i.e. remove / ban) a given root.

Note that I'm not adding a method for "undiscarding" a given root! That root will be banned indefinitely. That seems perfectly acceptable to me, but lmk if you have any concerns.

cc @LayneHaber

Reply:
Only thing is you need to be really careful while removing such that genuine roots are never removed by mistake. Otherwise I think it is fine

Reply:
Yeah this is a failure-recovery scenario, so we can ensure the correct roots are banned